packages/core-data/src/selectors.js

```diff
@@ -95,27 +95,62 @@ export const getUserQueryResults = createSelector(
 );
 
 /**
- * Returns whether the entities for the give kind are loaded.
+ * Returns the loaded entities for the given kind.
  *
+ * @deprecated since ??. Use getEntitiesConfig instead
  * @param {Object} state Data state.
  * @param {string} kind  Entity kind.
  *
  * @return {Array<Object>} Array of entities with config matching kind.
  */
 export function getEntitiesByKind( state, kind ) {
+	deprecated( "wp.data.select( 'core' ).getEntitiesByKind()", {
+		since: '6.0',
+		alternative: "wp.data.select( 'core' ).getEntitiesConfig()",
+	} );
+	return getEntitiesConfig( state, kind );
```

Comment:
these deprecations are so good. we have too many functions named different things that return the same data